# UVB_Melanoma
Little is known regarding the relationship between intratumor heterogeneity (ITH) and immune response in melanoma. Here we study this question in a novel, controlled experimental UVB mouse melanoma model that enables one to study the effects of intra-tumor heterogeneity on tumor aggressiveness and immune response independently of tumor mutational burden (TMB). The induction of UVB-derived mutations in parental melanoma cell-lines gives rise to high TMB tumors that are highly aggressive accompanied by decreased anti-tumor activity of tumor infiltrating lymphocytes (TILs). However, strikingly, UVB single-cell derived melanoma clones with similar high TMB levels but reduced ITH are swiftly rejected. Their rejection is accompanied by increased TIL reactivity, increased CD8+ T cell core infiltration and a less suppressive microenvironment. Using phylogenetic tree analyses and mixing experiments of 20 single cell UVB clones that lie along the phylogenetic tree we show that tumor rejection is inversely associated with the number of injected clones and their genetic diversity. Notably, similar observations are recapitulated and reinforced in the analysis of melanoma patient data, both in terms of overall melanoma patient survival and in response to immune check point therapy.  Taken together, our results highlight the importance of tumor ITH in melanoma and put forward the need to carefully quantify it to more accurately, when evaluating patient survival and response to checkpoint blockade. This repository provides the code for recapitulating mouse experimental results in TCGA patient data. 

# Methods description
From the TCGA data access portal (https://gdc-portal.nci.nih.gov/), we downloaded level 2 SNP array and germline + somatic variant call data (*.byallele.copynumber.data.txt, *.oxoG.snp.capture.tcga.vcf) for 432 skin cutaneous melanoma tumor and matched normal samples. Only 330 of these 432 had relevant clinical information to perform survival analyses. Across all 432 primary tumor samples, we apply CHAT (Li and Li, 2014) under default package settings to estimate tumor purity followed by estimation of cellular abundance of CNVs and somatic mutations from the SNP array and variant call data respectively. We found that the average sample purity estimated by CHAT is ~74% with only 14 samples having purity less than 25%. However, we do not pre-filter any of these samples in our downstream survival analyses as our final conclusions remain the same even after their removal. To estimate mutation load per sample, we count the number of somatic variant calls that were classified as missense or non-sense per sample. This data was obtained from the cbioportal website (https://www.cbioportal.org)To measure mutation load per sample, we simply count the number of somatic variant calls per sample. These calls include both silent + non-silent somatic mutations. Since CHAT detects CNVs using the circular binary segmentation algorithm (Olshen et al., 2004), which essentially partitions the genome into non-overlapping sections of same copy number, we estimate CNV load per sample in a manner similar to (Andor et al., 2016). To elaborate: for a tumor sample, let L_s be the length of a segment s of the genome and let CN_s be total copy number of that segment inferred by CHAT while considering the tumor purity. Let $X_s∈{0,1}$ be an indicator of deviation of CN_s from normal diploid copy number of 2. Then, CNV load is defined as:
![img](http://latex.codecogs.com/svg.latex?CNV load%3D%28%5Csum_%28s%3AX_s%3D1%29L_s%29%2F%28%5Csum_sL_s%29)

Given that tumor evolution is characterized by a series of clonal expansion events, we often find genetically distinct sub-populations of cells in the tumor with shared mutations or CNVs. The number of these genetically distinct subpopulations or clones is interpreted as the intra-tumor heterogeneity. Using CHAT, we can derive two estimates of number of clones by – clustering cellular abundances of somatic mutations (ITH1) or clustering cellular abundances of CNVs (ITH2). Both estimates convey important information of the underlying clonal structure at different resolutions. Hence, we set the overall intra-tumor heterogeneity of a sample as:
ITH=max⁡(ITH1,ITH2)
Given the limitation of a single bulk tumor sample per patient for inference, the above estimate is a lower bound and correlated with tumor purity (spearman’s rho = 0.232, p-value = 2.09E-5). 

In order to see the clinical relevance of the clone mixing experiments in mice we additionally also need to infer the diversity among clones in the same patient. This is done by re-constructing a phylogenetic tree over the clones. In this tree, each node represents a clone and each link represents a parent-child relationship. The root node represents the founder clone and the leaf nodes represent the clones observed at the time of the biopsy. Furthermore, for any parent clone i, the phylogenetic tree satisfies the following rule:
CCF_i≥ ∑_(j ∈ Children(i))▒〖CCF_j 〗
While it is impossible to know the correct phylogenetic tree describing the evolutionary history from a single biopsy, it is possible to estimate a posterior distribution over all possible phylogenetic trees. An algorithm to do that has been implemented previously and we use it in this work (https://github.com/ChrisMaherLab/ClonEvol). Given the posterior distribution, one can estimate the diversity of clones in patient data as follows:  For a tumor sample, let T be the set of top 50 plausible phylogenetic trees (ordered by their posterior probability). For a tree t ∈ T, let N(t) be the set of non-leaf nodes in that tree. The genetic diversity is then calculated as follows:
Diversity=1/50 ∑_(t∈T)▒〖1/(|N(t)|) ∑_(i∈N(t) )▒〖|Children(i)|〗〗

This quantity is also known as the branching factor of the tree. The higher this value, the more branches on average in the tree. Hence higher diversity of clones. In our analyses, we used both mutation and copy number data to infer distinct clones and their cellular abundances. Hence, the clonal diversity in the tumor sample was set as: max(Diversity estimated from mutation data, Diversity estimated from copy number data).

TCGA Survival Analysis 
To see if mutation load, CNV load, ITH or diversity are associated with overall patient survival, we stratified the TCGA patients into the following groups:
	Low mutation load (<= median of cohort), high mutation load (> median)
	Low cnv load (<= median), high cnv load (> median)
	Low ITH (<= median), High ITH (> median)
	Low diversity (<= median), high diversity (> median)

Given a cohort of patients, we then fit Kaplan-Meier survival curves for each group and their combinations and test if there are any significant survival differences between the groups using a Log-rank test. All survival analyses were performed using the survival and survcomp packages readily available for R. Due to potentially confounding effects of purity and other clinical factors, it is necessary to ascertain whether the observed associations with survival still hold after accounting for survival differences explained by other confounding factors. We performed a multivariate cox regression analysis in which patient age, tumor purity and clinical stage were included as additional factors. Our original conclusions do not change after running this analysis


# Analysis of the association between ITH, mutational load and patient survival across TCGA skin cutaneous melanoma samples 
A) Kaplan-Meier survival curves (time is measured in days on the X axis) of patients with high vs. low mutational load. Log rank statistic:1.96, p=0.16. B) Kaplan-Meier survival curves of patients with high vs. low CNV load Log rank statistic:0.31, p=577. C) Kaplan-Meier survival curves of patients with high vs. low ITH. Log rank statistic:3.97, p=0.046 D) Kaplan-Meier survival curve for patients segregated by number of clones. E) Kaplan-Meier survival curves of patients segregated based on the combination of mutational load and ITH. Log rank statistic:9.20, p=0.0267. F) Kaplan-Meier survival curves of patients segregated based on the combination of CNV load and ITH. Log rank statistic:4.57, p=0.206. G) CYT score of patients with high vs. low ITH. p-value = 7.6e-05, Wilcox’s rank sum test. H) CYT score of patients segregated by number of clones.  For further information regarding the analyses, contact sushantpatkar1994@gmail.com.

